File: 실습 코드/객체/Java10_Object/src/com/test/chap01_encapsulation/Score.java
package com.test.chap01_encapsulation;

public class Score {
		//필드
	
	private String name = "이창진";
	private int kor = 90;
	private int eng = 84;
	private int mat = 97;
	
	//생성자
	
	public Score() {}
	
	//총합을 구하여 출력하는 메소드
	public void sum() {
		int sum = kor+eng+mat;	
		System.out.println("총합 : " +sum);
	}
	
	//평균을 구하여 출력하는 메소드
	public void avg() {
		int avg = (kor+eng+mat)/3;
		System.out.println("평균 : "+avg);
	}
	
	//제일 높은 점수를 받은 과목을 찾는 메소드
	public void max() {
		int max = kor;
		if(max >= eng) {
					if(mat>= max) {
						max = mat;
					}
		}else {
			max =eng;	
			if(max <= mat) {
				 max = mat;
			}
			}
		System.out.println("최댓값 : "+max);
		}
	
/*     if(kor >eng && kor >mat){
 *              system.out.println("국어가 최댓값");
 *               else if(eng > kor && eng >mat){
 *               system.out.println("영어가 최댓값");
 *               }else{
 *               system.out.println("수학이 최댓값");
 *               }
 * 
 * 	
 */
	

	//제일 낮은 점수를 받은 과목을 찾는 메소드
	public void min() {
		int min = kor;
				if(min <= eng) {
						if(mat<= min) {
							min = mat;
						}
			}else {
				min =eng;	
				if(min >= mat) {
					 min = mat;
				}
				}
		System.out.println("최솟값 : "+min);		
				
	}
	//필드값을 확인 할 수 있는 information() 메소드
	public void information() {
		System.out.println("name : " + name + ", kor : " + kor + ", eng : "+eng + ", mat : " + mat);
		
		
	}
	
}



//괄호안에 쓰는 것은 왜 ?? 어떤 이유로 ??

File: 실습 코드/9. SERVLET/Jsp10_Ajax02/WebContent/score.js
function getParameterValues(){
	var name = "name=" + $("#name").val();
	var kor = "kor=" + $("#kor").val();
	var eng = "eng=" + $("#eng").val();
	var math = "math=" + $("#math").val();
	
	return "?"+name+"&"+kor+"&"+eng+"&"+math;
	
}


function load(){
	var url = "CalScore"+getParameterValues();
	
	httpRequest = new XMLHttpRequest(); //서버와 통신하는 객체
	httpRequest.onreadystatechange=callback;	//서버로 보낸 요청에 대한 응답을 받았을 때
												//어떤 동작을 할 것인지
	
	httpRequest.open("GET",url,true);		//true : 비동기통신
	httpRequest.send();
}

function callback(){
	alert("readystate: " + httpRequest.readyState);
	
	if(httpRequest.readyState == 4){
		alert("status: " + httpRequest.status);
		if(httpRequest.status=200){
			var obj = JSON.parse(httpRequest.responseText);
			$("#result").html(obj.name+"의 총점은"+obj.sum+"이고, 평균은 "+obj.avg+"입니다.");
		}else{
			alert("데이터를 계산할 수 없습니다.");
		}
		
	}

}

$(function(){
	$("#process").click(function(){
		load();
	});
});

/*
 readyState 
 1: loading - 로드 중
 2: loaded - 로드 됨      이부분쯤 toJSONString이 됨
 3: interactive - 통신 됨
 4: complete - 통신 완료    이 때 화면에 html 출력

-status
200 : 성공
404 : not found
500 : internal server error
 
 */



